semtypes/src/main/java/io/ballerina/types/typeops/ObjectOps.java

```diff
@@ -52,7 +54,15 @@ public SubtypeData complement(SubtypeData t) {
 
     @Override
     public boolean isEmpty(Context cx, SubtypeData t) {
-        return mappingSubtypeIsEmpty(cx, t);
+        return objectSubTypeIsEmpty(cx, t);
+    }
+
+    private static boolean objectSubTypeIsEmpty(Context cx, SubtypeData t) {
+        return memoSubtypeIsEmpty(cx, cx.mappingMemo, ObjectOps::objectBddIsEmpty, (Bdd) t);
+    }
+
+    private static boolean objectBddIsEmpty(Context cx, Bdd b) {
+        return bddEveryPositive(cx, b, null, null, MappingOps::mappingFormulaIsEmpty);
```

Comment:
I didn't get the difference between the old `return mappingSubtypeIsEmpty(cx, t);` line vs the new changes.

Reply:
We could also, extend `ObjectOps` class with `CommonOps` similar to other places and get rid of overrides to union, intersect, and diff() here.

Reply:
It is using `bddEveryPositive` instead of `bddEvery`in order to support distinct types